llvm/include/llvm/Analysis/IVDescriptors.h

```diff
@@ -423,6 +424,61 @@ class InductionDescriptor {
   SmallVector<Instruction *, 2> RedundantCasts;
 };
 
+/// A Conditional Scalar Assignment (CSA) is an assignment from an initial
+/// scalar that may or may not occur.
+class CSADescriptor {
```

Comment:
I don't think CAS is a very common term, would be good to have a more descriptive name if possible